hypervisor/util.go

```diff
@@ -0,0 +1,69 @@
+package hypervisor
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+type Volume struct {
+	Path    string
+	Format  string // raw|qcow2|...
+	AioType string // native|threads
+	Cache   string // none|unsafe|writethrough...
+}
+
+// parseVolumes parses --volume strings that are of following format:
+// --volume {volumePath}[:{options}]
+// Example: --volume /path/to/myvolume.img:format=raw:aio=native
+func ParseVolumes(volumeStrings []string) ([]Volume, error) {
+	res := []Volume{}
+	if volumeStrings == nil {
+		return res, nil
+	}
+	for _, volumeStr := range volumeStrings {
+		if v, err := parseVolume(volumeStr); err == nil {
+			res = append(res, *v)
+		} else {
+			return res, err
+		}
+
+	}
+	return res, nil
+}
+
+func parseVolume(volumeStr string) (*Volume, error) {
+	v := &Volume{
+		Path:    "",
+		Format:  "raw",
+		AioType: "native",
+		Cache:   "none",
+	}
+
+	for idx, part := range strings.Split(volumeStr, ":") {
+		if idx == 0 { // Volume path
+			if path, err := filepath.Abs(part); err == nil {
+				v.Path = path
+				continue
+			} else {
+				return nil, err
+			}
+		} else { // Volume settings
```

Comment:
No need for `else` here. The first `if` will either `continue` or break with an `error`.